evil-collection-ag.el

```diff
@@ -35,7 +35,7 @@
 
 (defun evil-collection-ag-setup ()
   "Set up `evil' bindings for `ag'."
-  (evil-define-key '(normal visual) ag-mode-map
+  (evil-collection-define-key '(normal visual) 'ag 'ag-mode-map
```

Comment:
Very minor but I think it's a little regrettable for the extra 'package' parameter when most keymaps (I know some don't) are 1:1 with the package name. (i.e. ag-mode-map -> ag).

Not a blocker or anything, just thought I'd comment.

Reply:
That's true for many packages, but certainly not all. See ivy, helm, pdf-tools, etc.

Another thought I had was to have a mapping somewhere between keymaps and conataining packages, which might be useful information to keep track of. 

Before you were using `evil-define-key` which uses some tricks to defer loading bindings until a map is defined (without needing to know the package), but I don't think it's a good idea to use this functionality for evil-collection. It can be tricky to debug when something goes wrong, and I think it's a good idea to be explicit about which keymap is where in case something changes upstream and it goes uncaught because `evil-define-key` hides what's going on in a sense.